### PR TITLE
[b] Python 2 now shows error message

### PR DESCRIPTION
**Pull request by catriuspham**
This issue explains this PR quite well https://github.com/EastAgile/robber.py/issues/36

### CODE REVIEW

File: robber/bad_expectation.py [hunk 1]
```diff
@@ -4,6 +4,7 @@ class BadExpectation(Exception):
     """
 
     def __init__(self, message):
+        super(BadExpectation, self).__init__(message)
```
Comment (tklarryonline) [inline robber/bad_expectation.py:7]: Woa, this simple line could take hours to figure...

**Comment by tklarryonline** [comment]
LGTM 👍

File: robber/bad_expectation.py (hunk 1 above)
Comment (catriuspham) [reply robber/bad_expectation.py:7]: Took me 5mins actually :| All I had to do was to search how to create custom `Exception` in Python 2.

Comment (catriuspham) [reply robber/bad_expectation.py:7]: Anyway, glad to see you survive until now :joy:

Comment (tklarryonline) [reply robber/bad_expectation.py:7]: Actually, calling `super()` when subclassing is a _de facto_ practice. It's just that it tends to slip through your eyes when reviewing the code...

**Comment by catriuspham** [comment]
@hieueastagile Just a tiny fix, how do you think?

**Comment by hieueastagile** [comment]
@catriuspham It looks good to me, but can we provide a test for it somehow?

**Comment by catriuspham** [comment]
@hieueastagile Just being curious, let take a look at `bad_expectation.py`:

```python
class BadExpectation(Exception):
    """
    Raised when an assertion fails.
    """

    def __init__(self, message):
        super(BadExpectation, self).__init__(message)
        self.message = message

    def __str__(self):
        return repr(self.message)
```

Take notice of the `repr` in the `__str__` function

With it the message will show like this:
```
BadExpectation: "Expected 1 to equal 2"
```

Without it the message will show like this:
```
BadExpectation: Expected 1 to equal 2
```

And I personally prefer the second one. Is there any other logic behind?

**Comment by tklarryonline** [comment]
The `repr` of a string is how the string being represented by the system. That's why you see the quotes.

For example:

```py
>>> a = 'Hello world'
>>> a
'Hello world'
>>> repr(a)
"'Hello world'"
>>> print(a)
Hello world
>>> print(repr(a))
'Hello world'
```

**Comment by catriuspham** [comment]
@tklarryonline Hehe thanks for the informatively informative information. I was just wondering if this `repr` had any effect on other parts of our library, or it was there just for personal taste.

**Comment by tklarryonline** [comment]
I really think that it's just there for the author's personal taste. I personally don't prefer using `repr`...

**Comment by catriuspham** [comment]
Well, I decided not to keep the `repr`. Doing this made the test look better. Anyway, in our new explanation system, the failure message will be divided into many lines, and we cannot do such with the presence of `repr`. Got any suggestions @hieueastagile?

**Comment by hieueastagile** [comment]
We can keep `str` only, `repr` is not needed in our purpose, I guess.

**Comment by catriuspham** [comment]
@oyster Please help merge this PR and close this issue https://github.com/EastAgile/robber.py/issues/36 thanks

**Comment by oyster** [comment]
Done & Done ;)

**Pull request merged**